joomla/joomla-cms
# [4.0] WebAsset for components, part 1

@wilsonge sorry had no time to make it before beta1, I hope it still okay. Other components I try to finish next week.

### Summary of Changes

This make components to use WebAsset,
Affected components:
```
com_content
    /administrator/index.php?option=com_content&view=articles
    /administrator/index.php?option=com_content&view=featured
    /administrator/index.php?option=com_content&view=article&layout=edit

com_categories
    /administrator/index.php?option=com_categories&view=categories&extension=com_content
    /administrator/index.php?option=com_categories&view=category&layout=edit&id=2&extension=com_content

com_associations
    /administrator/index.php?option=com_associations&view=associations
    /administrator/index.php?option=com_associations&view=associations&layout=modal
    /administrator/index.php?option=com_associations&view=association&layout=edit&itemtype=com_content.article&task=association.edit&id=70 (edit)

com_fields
    /administrator/index.php?option=com_fields&view=fields
    /administrator/index.php?option=com_fields&view=fields&layout=modal (popup)
    /administrator/index.php?option=com_fields&view=field&layout=edit&context=com_content.article

com_contact
    /administrator/index.php?option=com_contact&layout=modal
    /administrator/index.php?option=com_contact&view=contact&layout=edit

com_contenthistory
com_tags
com_users
com_banners
com_newsfeeds
```


### Testing Instructions

Apply patch, run `npm install`
Navigate around the site, try create/edit content in the component from the list
Inspect source code, you should see component scripts loaded.

### Expected result
All works


### Actual result
All works


ref #22435
